core/keyboard_nav/navigation.js

```diff
@@ -499,13 +500,11 @@ Blockly.navigation.connect_ = function(movingConnection, destConnection) {
   } else if (Blockly.navigation.moveAndConnect_(movingConnection, destConnection)){
     return true;
   } else {
-    try {
-      destConnection.checkConnection(movingConnection);
-    }
-    catch (e) {
-      // If nothing worked report the error from the original connections.
-      Blockly.navigation.warn_('Connection failed with error: ' + e);
-    }
+    var checker = movingConnection.getConnectionChecker();
```

Comment:
This part doesn't make sense, it used to be a try/check, now we're warning every time?

Reply:
This part used to be a pretty janky way of getting the error message out. It was only ever called when we knew that the connection would fail and an error would be thrown.

Reply:
Oh okay.

Reply:
Could we add a comment, as it does seem weird to assume the connection failed after checking it.